CONTRIBUTING.md instructions don't work on a fresh OS X install

I'm not sure whether it's an issue with the contributing instructions, our dependency management, or the Makefile, but something isn't right for someone that has just the bare minimum of things installed. If I try to run make, I get:

```shell
$ make build
Makefile:219: .bootstrap: No such file or directory
# cd /Users/alex/go/src/cloud.google.com/go; git rev-parse HEAD
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
HEAD
exit status 128
make: *** [.bootstrap] Error 1
```

My machine meets the stated prerequisites of the guide (C++ compiler > v6.0, go v1.6, git > v1.8, and bash) with the one exception that I have go 1.6.3 instead of 1.6.2. I believe my go environment is set up correctly:

```shell
$ echo $GOPATH
/Users/alex/go
$ pwd
/Users/alex/go/src/github.com/cockroachdb/cockroach
```

And I downloaded cockroach and its dependencies by running `go get -d github.com/cockroachdb/cockroach`, as instructed by the guide.

I'll see if I can figure out the problem and fix the guide.

--- a/GLOCKFILE
+++ b/GLOCKFILE
@@ -90,7 +90,7 @@ github.com/pkg/errors a22138067af1c4942683050411a841ade67fe1eb
 github.com/prometheus/client_model fa8ad6fec33561be4280a8f0514318c79d7f6cb6
 github.com/prometheus/common ebdfc6da46522d58825777cf1f90490a5b1ef1d8
 github.com/rcrowley/go-metrics bdb33529eca3e55eac7328e07c57012a797af602
-github.com/robfig/glock 62a8f79e21b6fe6d3e39436132596fd7a26c6518
+github.com/robfig/glock b8c84ff5ade15a6238ca61c20d3afc70d2e41276
 github.com/rubyist/circuitbreaker 7e3e7fbe9c62b943d487af023566a79d9eb22d3b
 github.com/russross/blackfriday 93622da34e54fb6529bfb7c57e710f37a8d9cbd8
 github.com/sasha-s/go-deadlock 09aefc0ac06ad74d91ca31acdb11fc981c159149

